### ISSUE TITLE
Upgrade to 16.1.3

### ISSUE THREAD

**Issue by mindbound** [plops/ecl-termux-binary#2]
According to the [release post](https://common-lisp.net/project/ecl/posts/ECL-1613-release.html), ECL 16.1.3 brings a number of new features, enhancements, and bugfixes, some of them significant. Please consider updating the binary if that's not too difficult.

**Comment by HJarausch** [comment]
You can build it yourself 

1. Unpack the source file into Termux' home directory (e.g.) and cd into it
2. Run the follwing commands there
`find . -type f -exec sed -i -e's|/bin/sh|/data/data/com.termux/files/usr/bin/bash|g' {} \;`
`sed -i -e's/stmeqia r2, {r0, r1}/stmiaeq r2, {r0, r1}/' src/libffi/src/arm/sysv.S`
3. `./configure --host=arm-linux-androideabi --with-system-gmp 
  --enable-boehm=included  --with-cxx --with-dffi --enable-shared=no 
  --prefix=/data/data/com.termux/files/usr/local`
4. make   or  make -j8 if you have 8 cores  - this takes some time
5. make install

Currently there is a minor glitch, i.e. a LINKER warning on startup.

**Comment by HJarausch** [comment]
pkg install termux-elf-cleaner     (if you haven't done so, already)

6. termux-elf-cleaner /data/data/com.termux/files/usr/local/bin/ecl

to get rid of the LINKER warning

**Comment by mindbound** [comment]
Thanks!

**Comment by KedalionDaimon** [comment]
Unfortunately, I am getting a type mismatch error when trying to "make" [EDIT: 16.1.13, on an ARM Cortex A53 processor]:

In file included from /data/data/com.termux/files/home/ecl/src/bdwgc/allchblk.c:17:               In file included from /data/data/com.termux/files/home/ecl/src/bdwgc/include/private/gc_priv.h:94:/data/data/com.termux/files/home/ecl/src/bdwgc/include/private/gcconfig.h:2035:19: error:               redeclaration of '_end' with a different         type: 'char []' vs 'int []'                      extern char _end[];                                          ^                              /data/data/com.termux/files/home/ecl/src/bdwgc/include/private/../gc.h:1622:30: note:                   previous declaration is here                 extern int __data_start[], _end[];                                          ^                   1 error generated.                               make[3]: * [Makefile:1527: allchblk.lo] Error 1make[3]: Leaving directory '/data/data/com.termux/files/home/ecl/build/gc'                        make[2]: * [Makefile:1710: install-recursive] Error 1                                           make[2]: Leaving directory '/data/data/com.termux/files/home/ecl/build/gc'                        make[1]: * [Makefile:190: libeclgc.a] Error 2  make[1]: Leaving directory '/data/data/com.termux/files/home/ecl/build'



I thought I could fix this by exchanging everywhere in ecl/src/bdwgc/include/private/gcconfig.h the string "char _end" with "int _end". However, now I am getting:

Building ecl_min...
.//libeclgc.a(pthread_stop_world.o): In function `android_thread_kill':
/data/data/com.termux/files/home/ecl/src/bdwgc/pthread_stop_world.c:452: undefined reference to `tkill'                                            /data/data/com.termux/files/home/ecl/src/bdwgc/pthread_stop_world.c:452: undefined reference to `tkill'                                            .//libeclgc.a(mach_dep.o): In function `GC_with_callee_saves_pushed':                             /data/data/com.termux/files/home/ecl/src/bdwgc/mach_dep.c:245: undefined reference to `getcontext'clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)               make[1]: * [Makefile:224: ecl_min] Error 1     make[1]: Leaving directory '/data/data/com.termux/files/home/ecl/build'                           make: * [Makefile:72: all] Error 2


Regarding this error, I found: https://github.com/android-ndk/ndk/issues/258

- basically, this says it is the Boehm GC library.

EDIT: I tried to use the Boehm GC library from Github, but no dice - configure fails:

https://github.com/ivmai/bdwgc/tree/3ca60bd5dcb0e1c04fdc6459417824c05b292627

It would be very nice of you if you could update this, as the build description, as is, does not work.

**Comment by its-pointless** [comment]
Iv got working ecl as a pull request for termux-packages on all 4 supported archs.

**Comment by plops** [comment]
I just compiled the following version for aarch64 and added the binary to the repo:
ECL (Embeddable Common-Lisp) 16.1.3 (git:06f0a93421b261aaa0fbfb2c9d077eebfc366280)
Copyright (C) 1984 Taiichi Yuasa and Masami Hagiya
Copyright (C) 1993 Giuseppe Attardi
Copyright (C) 2013 Juan J. Garcia-Ripoll
Copyright (C) 2018 Daniel Kochmanski
Copyright (C) 2019 Daniel Kochmanski and Marius Gerbershagen
